services/document-manager/backend/app/docman/utils/document_upload_helper.py

```diff
@@ -285,3 +292,145 @@ def complete_upload(cls, key, new_key, doc_guid, versions=None, version_guid=Non
+
+                if response.status_code == 200:
+                    current_app.logger.info(f"Successfully downloaded document: {doc.file_display_name}")
+                    file_data = response.get_data()
+
+                    zipf.writestr(doc.file_display_name, file_data)
+                else:
+                    raise Exception(f"Failed to download document: {doc.file_display_name}")
+
+        return shpz_path
+    @classmethod
+    def download_kml_kmz_files(cls, doc):
+        # Download the file from object store
+        response = ObjectStoreStorageService().download_file(doc.object_store_path, doc.file_display_name, False)
+
+        # Check the response status
+        if response.status_code == 200:
+            file_data = response.get_data()  # obtain file data from response body
+            with open(doc.file_display_name, 'wb') as file:
+                file.write(file_data)
+        else:
+            raise FileNotFoundError(f"Failed to download file: {doc.file_display_name}")
+
+    @classmethod
+    def send_spatial_file_to_geomark(cls, file_path):
+        file_extension = os.path.splitext(file_path)[1].lstrip('.')
+
+        # Ensure the format is one of the allowed ones
+        assert file_extension in ['shpz', 'kml', 'kmz'], "Invalid file format. Must be one of: 'shpz', 'kml', 'kmz'."
+
+        mime_type = mimetypes.guess_type(file_path)[0]
+
+        url = 'https://apps.gov.bc.ca/pub/geomark/geomarks/new'
+
+        with open(file_path, 'rb') as file:
+            multipart_data = MultipartEncoder(
+                fields={
+                    'body': (os.path.basename(file_path), file, mime_type),
+                    'resultFormat': 'json',
+                    'format': file_extension,
+                    'srid': '4326'
+                }
+            )
+            headers = {
+                'Content-Type': multipart_data.content_type
+            }
+
+            # Send the request
+            response = requests.post(
+                url,
+                data=multipart_data,
+                headers=headers
+            )
+
+        # Delete the .shpz file
+        os.remove(file_path)
+
+        return response.json() if response.text.strip() else None
+
+    @classmethod
+    def complete_bundle_upload(cls, bundle_document_guids, name):
+        bundle_documents = Document.find_by_document_guid_many(bundle_document_guids)
+
+        if len(bundle_documents) != len(bundle_document_guids) or not bundle_documents:
+            raise NotFound('One or more documents not found')
+
+        cls.validate_bundle(bundle_documents)
+
+        if len(bundle_documents) > 1:
+            file_path = f'{name}.shpz'
+            cls.zip_spatial_files(bundle_documents, name)
+        else:
+            file_path = bundle_documents[0].file_display_name
+            cls.download_kml_kmz_files(bundle_documents[0])
+
+        geomark_response = cls.send_spatial_file_to_geomark(file_path)
```

Comment:
I see there's a cleanup call to remove the downloaded file at the end of the `send_spatial_file_to_geomark` method - however I think there's still a chance that the file is not removed if the call to geomark fails. I'm wondering if we could wrap this in a `try / finally` block to make sure it gets cleaned up regardless of the outcome.

```python
try:
    geomark_response = cls.send_spatial_file_to_geomark(file_path)
finally:
    os.remove(file_path)
```